jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenSettingsServerCredentials.java

```diff
@@ -32,31 +38,50 @@
   @VisibleForTesting static final String CREDENTIAL_SOURCE = "Maven settings";
 
   private final Settings settings;
+  @Nullable private final SettingsDecrypter settingsDecrypter;
 
-  MavenSettingsServerCredentials(Settings settings) {
+  MavenSettingsServerCredentials(Settings settings, @Nullable SettingsDecrypter settingsDecrypter) {
     this.settings = settings;
+    this.settingsDecrypter = settingsDecrypter;
   }
 
   /**
    * Attempts to retrieve credentials for {@code registry} from Maven settings.
    *
    * @param registry the registry
    * @return the credentials for the registry
+   * @throws MojoExecutionException if the credentials could not be retrieved
    */
   @Nullable
-  RegistryCredentials retrieve(@Nullable String registry) {
+  RegistryCredentials retrieve(@Nullable String registry) throws MojoExecutionException {
     if (registry == null) {
       return null;
     }
 
-    Server registryServerSettings = settings.getServer(registry);
-    if (registryServerSettings == null) {
+    Server registryServer = settings.getServer(registry);
+    if (registryServer == null) {
       return null;
     }
 
+    if (settingsDecrypter != null) {
+      SettingsDecryptionRequest request = new DefaultSettingsDecryptionRequest(registryServer);
+      SettingsDecryptionResult result = settingsDecrypter.decrypt(request);
+      // un-encrypted passwords are passed through, so a problem indicates a real issue
+      for (SettingsProblem problem : result.getProblems()) {
+        if (problem.getSeverity() == SettingsProblem.Severity.ERROR
+            || problem.getSeverity() == SettingsProblem.Severity.FATAL) {
+          throw new MojoExecutionException(
+              "Unable to decrypt password for " + registry + ": " + problem);
+        }
+      }
+      if (result.getServer() != null) {
+        registryServer = result.getServer();
+      }
```

Comment:
If it is `null`, should we error?

Reply:
`SettingsDecrypter` and `SettingsDecryptionResult` are under-documented and does not discuss the possible meanings of the return results.  `SettingsDecryptionResult#getServers()` does note that:
```java
    /**
     * Gets the decrypted servers.
     *
     * @return The decrypted server, can be empty but never {@code null}.
     */
    List<Server> getServers();
```
and the implementation of `getServer()` returns the first element from `getServers()` and `null` if there are no servers.

By this point, we have no errors reported from the decryption request, but we have no servers.  So one interpretation is: no decryption was required, so use the original server.